site/zenodo_rdm/legacy/serializers/schemas/common.py

```diff
@@ -71,7 +71,10 @@ class CreatorSchema(Schema):
     def dump_affiliation(self, obj):
         """Dump affiliation."""
         if obj.get("affiliations"):
-            return obj["affiliations"][0]["name"]
+            aff = obj["affiliations"][0]
+            # One of 'name' or 'id' must be present
+            name_or_id = aff.get("name") or aff.get("id")
```

Comment:
I am not sure if we can live only with the `id`. 
However assuming `name` is always there is wrong (it's only there for custom ones).

Ping @slint